service/warehouse/model/service.go

```diff
@@ -44,10 +45,25 @@ type Warehouse struct {
 	Name        string `json:"name" validate:"required"`
 	Type        string `json:"type" validate:"required"`
 	Address     string `json:"address" validate:"required"`
+	Racks       []Rack `json:"racks"`
 }
 
 type GetWarehousesRequest struct {
 	BranchId string `json:"branch_id" validate:"required"`
 	Keyword  string `json:"keyword"`
 	Type     string `json:"type"`
 }
+
+type WarehouseRack struct {
+	WarehouseId string `json:"warehouse_id" validate:"required"`
+	RackId    string `json:"branch_id" validate:"required"`
+}
+
+type AddWarehouseRequest struct {
+	BranchId string   `json:"branch_id" validate:"required"`
```

Comment:
ini isi nya AddWarehouseRequest dikeluarin ke model `type Warehouse struct { ... }` trus itu yang dipake di transaction sama di impl nya

Reply:
maksudnya gimana ya? saya ndak paham

Reply:
```
type Warehouse struct {
        BranchId string `json:"branch_id" validate:"required"`
        Code     string   `json:"code" validate:"required"`
	Name     string   `json:"name" validate:"required"`
	Type     string   `json:"type" validate:"required"`
	Address  string   `json:"address" validate:"required"`
	RackIds  []string `json:"rack_ids" validate:"required"`
}
```

```
type AddWarehouseRequest struct {
        Warehouse
}
```

Reply:
Itu beda mas, bagian racknya klo warehouse pake model Rack, kalau request pakai RackIds

Reply:
itu contoh, kalau detail nya u atur sendiri, yg penting u get the point.